cli/replicaset/cconfig.go

```diff
@@ -278,19 +278,77 @@ loop:
 	if len(unfound) > 0 {
 		msg := fmt.Sprintf("could not connect to: %s", strings.Join(unfound, ","))
 		if !ctx.Force {
-			return fmt.Errorf("all instances in the target replicast should be online, %s", msg)
+			return fmt.Errorf("all instances in the target replicaset should be online, %s", msg)
 		}
 		log.Warn(msg)
 	}
 
-	isConfigPatched, err := c.promote(targetInstance.InstanceCtx, ctx)
+	isConfigPublished, err := c.promote(targetInstance, ctx)
+	// Firstly, we check if the config was published.
+	if isConfigPublished {
+		err = errors.Join(err, reloadCConfig(instances))
+	}
 	if err != nil {
 		return err
 	}
-	if isConfigPatched {
-		err = reloadCConfig(instances)
+	return nil
+}
+
+// Demote demotes an instance in the application.
+func (c *CConfigApplication) Demote(ctx DemoteCtx) error {
+	replicasets, err := c.Discovery()
+	if err != nil {
+		return fmt.Errorf("failed to get replicasets: %w", err)
 	}
-	return err
+	var (
+		targetReplicaset Replicaset
+		targetInstance   Instance
+		found            bool
+	)
+loop:
+	for _, replicaset := range replicasets.Replicasets {
+		for _, instance := range replicaset.Instances {
+			if instance.Alias == ctx.InstName {
+				targetReplicaset = replicaset
+				targetInstance = instance
+				found = true
+				break loop
+			}
+		}
+	}
```

Comment:
It could be a method in the `libcluster` to found an instance with group name, replicaset name.

Reply:
I just see these loops spreading throughout the code.

https://github.com/tarantool/tt/blob/748dc838a8b378542b9d4580dcbfae091a4b1f01/cli/replicaset/cconfig.go#L251-L261

Reply:
Helpers were added. Thank you!